src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -0,0 +1,937 @@
+                set_kv = KeyValue(
+                    key=key,
+                    label=label,
+                    value=json.dumps(
+                        feature_flag_value,
+                        default=lambda o: o.__dict__),
+                    content_type=FEATURE_FLAG_CONTENT_TYPE,
+                    tags=retrieved_kv.tags if retrieved_kv.tags else tags)
+                set_kv.etag = retrieved_kv.etag
+                set_kv.last_modified = retrieved_kv.last_modified
+
+            # Convert KeyValue object to required FeatureFlag format for
+            # display
+            feature_flag = map_keyvalue_to_featureflag(
+                set_kv, show_conditions=True)
+            entry = json.dumps(feature_flag, default=lambda o: o.__dict__, indent=2, sort_keys=True)
+
+        except Exception as exception:
+            # Exceptions for ValueError and AttributeError already have customized message
+            # No need to catch specific exception here and customize
+            raise CLIError(str(exception))
+
+        confirmation_message = "Are you sure you want to set the feature flag: \n" + entry + "\n"
+        user_confirmation(confirmation_message, yes)
+
+        try:
+            updated_key_value = azconfig_client.add_keyvalue(
+                set_kv,
+                ModifyKeyValueOptions()) if set_kv.etag is None else azconfig_client.update_keyvalue(
+                    set_kv,
+                    ModifyKeyValueOptions())
+            return map_keyvalue_to_featureflag(
+                keyvalue=updated_key_value, show_conditions=True)
+        except HTTPException as exception:
+            if exception.status == StatusCodes.PRECONDITION_FAILED:
+                logger.debug(
+                    'Retrying setting %s times with exception: concurrent setting operations',
+                    i + 1)
+                time.sleep(retry_interval)
+            else:
+                raise CLIError(str(exception))
+        except Exception as exception:
+            raise CLIError(str(exception))
+    raise CLIError(
+        "Failed to set the feature flag '{}' due to a conflicting operation.".format(feature))
+
+
+def delete_feature(cmd,
+                   feature,
+                   name=None,
+                   label=None,
+                   yes=False,
+                   connection_string=None):
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+
+    delete_one_version_message = "Are you sure you want to delete the feature '{}'".format(
+        feature)
+    confirmation_message = delete_one_version_message
+    user_confirmation(confirmation_message, yes)
+
+    try:
+        retrieved_keyvalues = __list_all_keyvalues(azconfig_client,
+                                                   feature=feature,
+                                                   label=label if label is not None else "")
```

Comment:
>"" [](start = 89, length = 2)

nit: although we do treat empty string as null label in list, can you use QueryKeyValueCollectionOptions.empty_label here?